explainer.md

```diff
@@ -1,121 +1,159 @@
+If you write an email, instant message, or simply favourite a tweet, the application needs to communicate that data to the server. If that fails, either due to user connectivity, service availability or anything in-between, the app can store that action in some kind of 'outbox' for retry later.
 
-The web currently lacks any ability to provide this sort of functionality in a user-friendly or power-efficient way. Current approaches require an application (or tab) to be running and rely on slow, battery-intensive pings.
+Unfortunately, on the web, that outbox can only be processed while a page on the site is running. This is particularly problematic on mobile, where browsing contexts are frequently shut down to free memory.
 
-Native application platforms provide  [job scheduling](https://developer.android.com/reference/android/app/job/JobScheduler.html) APIs that enable developers to collaborate with the system to ensure low power usage and background-driven processing. The web platform needs capabilities like this too.
+Native application platforms provide [job scheduling](https://developer.android.com/reference/android/app/job/JobScheduler.html) APIs that enable developers to collaborate with the system to ensure low power usage and background-driven processing. The web platform needs capabilities like this too.
 
-We propose a new API that extends [Service Workers](https://github.com/slightlyoff/ServiceWorker) with a new `onsync` event and a new API for registering (and unregistering) interest in `onsync`. Together, these APIs form the basis of a powerful new capability for rich web apps.
+We propose a new API that extends [Service Workers](https://github.com/slightlyoff/ServiceWorker) with a `sync` event and an API for signalling the desire for this event to fire.
 
-## Use Cases
+### The API
 
-There are two general use cases that the `onsync` event is designed to address:
+**To request a sync:**
 
-1. Notification when next online to upload new content (email, docs, tweets, saved state)
-2. Periodic synchronization opportunities (static resources, content updates (e.g., docs, articles email), logging)
+```js
+navigator.serviceWorker.ready.then(function(registration) {
+  registration.sync.register({
+    tag: 'outbox' // default: ''
+  }).then(function(syncReg) {
+    // success
+  }, function() {
+    // failure
+  })
+});
+```
 
-In both cases the event will fire _even if the browser is currently closed_, though it may be delayed, see the description of the register function below.
+* `tag`: This operates like a notification's tag. If you register a sync and an existing sync with the same tag is pending, it returns the existing registration. If further options are added to `.register`, the existing registration will be updated.
```

Comment:
What does adding further options to `register` look like? Since all of the options have defaults, does this mean that we track which ones were specified explicitly, and which were left as default? Or does this just mean that `register({tag:tag})` can either act like `register()` or `getRegistration`, depending on whether the tag is already registered?